src/connector/src/source/nats/source/reader.rs

```diff
@@ -85,8 +87,81 @@ impl SplitReader for NatsSplitReader {
                 properties.stream.clone(),
                 split_id.to_string(),
                 start_position.clone(),
+                properties.durable_name.clone(),
+                properties.description.clone(),
+                properties.ack_policy.clone(),
+                properties.ack_wait.clone().map(|s| {
+                    Duration::from_secs(s.parse::<u64>().expect("failed to parse ack_wait to u64"))
+                }),
+                properties.max_deliver.clone().map(|s| {
+                    s.parse::<i64>()
+                        .expect("failed to parse max_deliver to i64")
+                }),
+                properties.filter_subject.clone(),
+                properties
+                    .filter_subjects
+                    .clone()
+                    .map(|s| s.split(',').map(|s| s.to_string()).collect()),
```

Comment:
These don't look good. We'd better parse the config when creating the properties. So that `CREATE SOURCE` will reject invalid config.

e.g., using `#[serde_as(as = "Option<DisplayFromStr>")]` or `#[serde(deserialize_with = ...)]`

Reply:
How would the definition would look like? Simply `[10.30.60]` for instance?

Reply:
Ah indeed that seems much better, even without understanding the error related to config parsing that parsing seemed a bit sketchy. I’ll give it a try in the upcoming days @xxchan thank you

Reply:
I've moved the consumer specific parameters into a separate struct, and added some typed properties based on @xxchan's suggestion